private/astarte_linked_list.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * (C) Copyright 2018-2023, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR Apache-2.0
+ */
+
+/**
+ * @file astarte_linked_list.h
+ * @brief Utility module containing a basic linked list implementation.
+ *
+ * @details This library does not perform deep copies when storing values.
+ * The user should ensure that any value is freed correctly when appropriate.
+ */
+
+#ifndef _ASTARTE_LINKED_LIST_H_
+#define _ASTARTE_LINKED_LIST_H_
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "astarte.h"
+
+typedef struct
+{
+    struct astarte_linked_list_node *head;
+    struct astarte_linked_list_node *tail;
+} astarte_linked_list_handle_t;
+
+typedef struct
+{
+    astarte_linked_list_handle_t *handle;
+    struct astarte_linked_list_node *node;
+} astarte_linked_list_iterator_t;
+
+typedef struct
+{
+    void *value;
+    size_t value_len;
```

Comment:
In some use cases, it might be necessary. For example, if we want to store some binary data for later use, storing just the pointer without its size would make it unusable.
In addition, creating an `astarte_linked_list_find` method would be impossible without the size of the stored data.
Which use case do you have in mind in which it would be unavailable?